tests/integ/sagemaker/serve/test_serve_hf_dlc.py

```diff
@@ -0,0 +1,102 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import pytest
+
+from sagemaker.serve.builder.schema_builder import SchemaBuilder
+from sagemaker.serve.builder.model_builder import ModelBuilder, Mode
+from sagemaker.serve.utils.types import ModelServer
+
+from tests.integ.sagemaker.serve.constants import (
+    HF_DIR,
+    PYTHON_VERSION_IS_NOT_310,
+    SERVE_SAGEMAKER_ENDPOINT_TIMEOUT,
+)
+
+from tests.integ.timeout import timeout
+from tests.integ.utils import cleanup_model_resources
+import logging
+
+logger = logging.getLogger(__name__)
+
+sample_input = {
+    "inputs": "The man worked as a [MASK].",
+}
+
+loaded_response = [{'score': 0.0974755585193634, 'token': 10533, 'token_str': 'carpenter', 'sequence': 'the man worked as a carpenter.'},
+                   {'score': 0.052383411675691605, 'token': 15610, 'token_str': 'waiter', 'sequence': 'the man worked as a waiter.'},
+                   {'score': 0.04962712526321411, 'token': 13362, 'token_str': 'barber', 'sequence': 'the man worked as a barber.'},
+                   {'score': 0.0378861166536808, 'token': 15893, 'token_str': 'mechanic', 'sequence': 'the man worked as a mechanic.'},
+                   {'score': 0.037680838257074356, 'token': 18968, 'token_str': 'salesman', 'sequence': 'the man worked as a salesman.'}]
+
+
+@pytest.fixture
+def input():
+    return {"inputs": "The man worked as a [MASK]."}
+
+
+@pytest.fixture
+def model_builder_model_schema_builder():
+    return ModelBuilder(
+        model_path=HF_DIR,
+        model="bert-base-uncased",
+        schema_builder=SchemaBuilder(sample_input, loaded_response),
+        image_uri=
+        "763104351884.dkr.ecr.us-east-1.amazonaws.com/huggingface-pytorch-inference:1.7.1-transformers4.6.1-cpu-py36-ubuntu18.04",
+        model_server=ModelServer.HUGGINGFACE_DLC,
```

Comment:
Not clear, why HUGGINGFACE_DLC is categorized as model server.

Reply:
That's a good call out, and HF transformers DLC uses multi-model-server: https://github.com/aws/deep-learning-containers/blob/master/huggingface/pytorch/inference/docker/2.1/py3/Dockerfile.cpu#L155.

I think we should not break the contract of the enum value unless it is actually introducing a new model server type.

Reply:
Ack. @samruds we may have to add additional logic as well:
i.e 
```
if specified model server is MMS and model is string and model md is retrievable from HuggingFace Hub:
    -> deploy with HF DLC
```

Reply:
if specified model server is MMS and model is string and model md is retrievable from HuggingFace Hub:
    -> deploy with HF DLC -> Is it okay to take this as a follow up because it is introducing additional functionality that is not necessarily needed for milestone 1.